### DISCUSSION TITLE
rule for button and 2 relays?

### DISCUSSION THREAD

**Discussion by smitec11** [arendst/Tasmota#14308] [Q&A] [answered]
Hi All,

probably a very simple question, but I can not figure it out.
I've got a button and two relays :

Relay2 should be activated if the button is pressed and Relay1 is active.
(Relay1 is controlled externally by domoticz, so used as an enable for Relay2)

anyone can tell me how to do this?

regards,
Chris

**Comment by barbudor** [comment, marked as answer]
You don't define every stage change so difficult to provide an answer
- what should happen if relay 1 goes off while relay 2 is on
- what should happen when button is  pressed and relay 2 is on

**Reply by smitec11** [reply]
here's the table :

Relay1 Button1 Result_Relay2
0 0  0
0 1  0
1 0  0
1 1  1

Relay2 must only be active is relay1 is active and button is active.

**Comment by smitec11** [comment]
here's what I got now 👍 
relay 1 and 2 are swapped from my original setup

SwitchMode 2         -> because switch is inverted

Rule1 
ON Power2#State DO var1 %value% ENDON
ON Switch1#State Power1 %var1% ENDON
Rule1 1

I see that Power2 sets the varable, but switch1  toggle the relay1, not follow state of switch

anyone an idea?

**Comment by sfromis** [comment]
Looks a lot like the [same issue](https://discord.com/channels/479389167382691863/790182324939128832/930093498227638292) discussed on Discord... Better not cross-post.

**Comment by smitec11** [comment]
got it 👍 

SwitchMode1 2         -> because switch is inverted

Rule1 
ON Power2#State DO var1 %value% ENDON
ON Switch1#State DO Power1 %var1% ENDON
ON Switch1#State=0 DO Power1 0 ENDON

Rule1 1
